packages/m2-theme/public/index.php

```diff
@@ -31,7 +31,7 @@
 
             // Multistore
             window.storeList = JSON.parse(`<?= $this->getStoreListJson() ?>`);
-            window.storeRegexText = `/(${window.storeList.join('|')})?`;
+            window.storeRegexText = `/(${window.storeList.join('|')})/?`;
```

Comment:
This change makes the store regex stricter, because now the group in parens is not optional. Please, keep it optional to be consistent with previous behavior. I think this may break when you have multistore on/off, requires additional testing.